docs/design/fabric-v2+/fpc-key-dist.puml

```diff
@@ -180,125 +179,147 @@ group chaincode key distribution
   return true /'Enclave2 -> TLCC_Enclave2'/
   note left
     if false, abort
   end note
-  Enclave2      -> Enclave2         :   extract Enclave1_VK from cckeydist_start_message.key_distribution_message
-  Enclave2      -> ERCC2            ++: GetRegistration(Enclave1_VK)
+  Enclave1      -> Enclave1         :   extract Enclave1_VK from cckeydist_start_message.key_distribution_message
+  Enclave1      -> ERCC1            ++: GetRegistration(Enclave1_VK)
   return Registered_Enclave1_Credentials /'Enclave2 -> ERCC2'/
-  Enclave2      -> TLCC_Enclave2    ++: check_ercc_committed_data(Registered_Enclave1_Credentials)
+  Enclave1      -> TLCC_Enclave1    ++: check_ercc_committed_data(Registered_Enclave1_Credentials)
   return true /'Enclave2 -> TLCC_Enclave2'/
   note left
     if false, abort
   end note
-  Enclave2      -> Enclave2         :   check Enclave1_VK matches in registration_message and cckeydist_start_message
-  Enclave2      -> Enclave2         :   check CC_Id in Registered_Enclave1_Credentials matches internal one
-  note over Enclave2
+  Enclave1      -> Enclave1         :   check Enclave1_VK matches in registration_message and cckeydist_start_message
+  Enclave1      -> Enclave1         :   check CC_Id in Registered_Enclave1_Credentials matches internal one
+  note right Enclave1
     Since Enclave1's Credentials are committed, then Enclave1's Evidence has already been verified by ERCC.
     Hence, verifying Enclave1's Evidence at this point does not appear necessary.
   end note
-  Enclave2      -> Enclave2         :   verify Sig_Enclave1
-  Enclave2      -> Enclave2         :   SEK, Chaincode_DK <- decrypt(cckeydist_start_message.encrypted_keys_message) with Enclave2_DK
-  Enclave2      -> Enclave2         :   extract Chaincode_EK from Chaincode_DK
-  note over Enclave2
-    TODO: seal keys
-  end note
-  Enclave2      -> Enclave2         :   Sig_Enclave2 <- sign Chaincode_EK with Enclave2_SK
-  Enclave2      -> Enclave2         :   cckey_registration_message <- <Enclave2_VK, Sig_Enclave2, Chaincode_EK>
-
-  return cckey_registration_message, /'Enclave2 -> ECC2'/
-  return cckey_registration_message, /'ECC2 -> Peer2_CLI'/
-
-  return cckey_registration_message /'Admin2 -> Peer2_CLI'/
-
-  ref over ERCC1, Admin2, ERCC2, Orderer
-    chaincode key registration (same as that from Admin1)
-  end ref
+  Enclave1      -> Enclave1         :   verify Sig_Enclave1
+  Enclave1      -> Enclave1         :   SEK, Chaincode_DK <- decrypt(cckeydist_start_message.encrypted_keys_message) with Enclave2_DK
+  Enclave1      -> Enclave1         :   extract Chaincode_EK from Chaincode_DK
+  Enclave1 -> Enclave1 : sealed_cckeys <- seal <SEK, Chaincode_EK, Chaincode_DK>
+  Enclave1      -> Enclave1         :   SIG_Enclave2 <- sign Chaincode_EK with Enclave2_SK
+  Enclave1      -> Enclave1         :   cckey_registration_message <- <Enclave2_VK, Sig_Enclave2, Chaincode_EK>
+
+  return cckey_registration_message /'Enclave2 -> ECC2'/
```

Comment:
You have to return sealed_cckeys and store it here.
BTW: given that we have now two separate sealed blobs, would be nice to mention both explicit and with same name as here and in regtration  in fpc-cc-invocation?

Reply:
right! good catch!